scripts/translation/lib/RevtagParser.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ *  +----------------------------------------------------------------------+
+ *  | Copyright (c) 1997-2023 The PHP Group                                |
+ *  +----------------------------------------------------------------------+
+ *  | This source file is subject to version 3.01 of the PHP license,      |
+ *  | that is bundled with this package in the file LICENSE, and is        |
+ *  | available through the world-wide-web at the following url:           |
+ *  | https://www.php.net/license/3_01.txt.                                |
+ *  | If you did not receive a copy of the PHP license and are unable to   |
+ *  | obtain it through the world-wide-web, please send a note to          |
+ *  | license@php.net, so we can mail you a copy immediately.              |
+ *  +----------------------------------------------------------------------+
+ *  | Authors:     André L F S Bacci <ae php.net>                          |
+ *  +----------------------------------------------------------------------+
+ *  | Description: Parse revision and credits from XML comments.           |
+ *  +----------------------------------------------------------------------+
+ */
+
+require_once __DIR__ . '/all.php';
+
+class RevtagInfo
+{
+    public string $revision = "";
+    public string $maintainer = "";
+    public string $status = "";
+    public string $credits = "";
+    public array  $errors = [];
+}
+
+class RevtagParser
+{
+    static function parseInto( string $lang , RevcheckFileList & $list )
+    {
+        foreach( $list->list as $entry )
+            $entry->revtag = RevtagParser::parseFile( $lang . '/' . $entry->file );
+    }
+
+    public static function parseFile( string $filename ): RevtagInfo|null
+    {
+        $doc = XmlUtil::loadFile( $filename );
+        $ret = new RevtagInfo;
+        RevtagParser::parseNodeRecurse( $doc , $ret , $filename );
+        return $ret;
+    }
+
+    public static function parseText( string $contents ): RevtagInfo|null
+    {
+        $doc = XmlUtil::loadText( $contents );
+        $ret = new RevtagInfo;
+        RevtagParser::parseNodeRecurse( $doc , $ret );
+        return $ret;
+    }
+
+    public static function parseNodeRecurse( DOMNode $node , RevtagInfo & $ret , $filename = "" )
+    {
+        if ( $node->nodeType == XML_COMMENT_NODE )
+            RevtagParser::parseComment( $node , $ret , $filename );
+
+        foreach( $node->childNodes as $child )
+            RevtagParser::parseNodeRecurse( $child , $ret , $filename );
+    }
+
+    public static function parseComment( DOMNode $node , RevtagInfo & $ret , $filename = "" )
```

Comment:
I have two questions and I will use this line as an example.

1. Why so many of the new functions do not specify the return type? 
2. Why are you using pass-by-ref? At least in this example the pass-by-ref seems completely unnecessary. I don't think you ever need to use pass-by-ref, you can just return what you need to return.

Reply:
1. No specific reason. Feel free to patch every and all functions with their expected return types.
2. This is possibly a vestige from my local personal scripts, when they used primitives values as "return accumulators". With class this is not necessary. I may test this to remove the pass by reference syntax.